shared/AppInsightsCore/Tests/Unit/src/CookieManager.Tests.ts

```diff
@@ -749,6 +774,513 @@ export class CookieManagerTests extends AITestClass {
+            test: () => {
+                // Create a dynamic config that can be updated
+                let configValues = {
+                    cookieCfg: objExtend({}, this._cookieMgrCfg, {
+                        enabled: false
+                    })
+                };
+                let manager = createCookieMgr(configValues);
+
+                let newKey = "test." + newId();
+                let newValue = newId();
+
+                // Set a cookie while disabled - should be cached
+                let result = manager.set(newKey, newValue);
+                Assert.equal(true, result, "Set should return true even when disabled (cached)");
+                Assert.equal(undefined, this._testCookies[newKey], "Cookie should not be in storage when disabled");
+
+                // Get should return the cached value
+                Assert.equal(newValue, manager.get(newKey), "Should return cached value when disabled");
+
+                // Enable cookies via dynamic config change - should flush cached values
+                configValues.cookieCfg.enabled = true;
+
+                // Verify the cookie was flushed to actual storage
+                Assert.equal(newValue + "; path=/", this._testCookies[newKey], "Cookie should be flushed to actual storage via config change");
+                Assert.equal(newValue, manager.get(newKey), "Should still return the value after enabling");
+            }
+        });
+
+        this.testCase({
+            name: "CookieManager: Dynamic config change to enable cookies with multiple cached values",
+            test: () => {
+                // Create a dynamic config that can be updated
+                let configValues = {
+                    cookieCfg: objExtend({}, this._cookieMgrCfg, {
+                        enabled: false
+                    })
+                };
+                let manager = createCookieMgr(configValues);
+
+                let cookies = [
+                    { key: "test1." + newId(), value: newId() },
+                    { key: "test2." + newId(), value: newId() },
+                    { key: "test3." + newId(), value: newId() }
+                ];
+
+                // Set multiple cookies while disabled - should be cached
+                cookies.forEach(cookie => {
+                    let result = manager.set(cookie.key, cookie.value);
+                    Assert.equal(true, result, "Set should return true even when disabled (cached)");
+                    Assert.equal(undefined, this._testCookies[cookie.key], "Cookie should not be in storage when disabled");
+                    Assert.equal(cookie.value, manager.get(cookie.key), "Should return cached value");
+                });
+
+                // Enable cookies via dynamic config change - should flush all cached values
+                configValues.cookieCfg.enabled = true;
+
+                cookies.forEach(cookie => {
```

Comment:
You need to add a clock.tick(1) after changing the state to allow the dynamic configurations changes to take affect